lib/fluent/registry.rb

```diff
@@ -43,11 +46,18 @@ def lookup(type)
       raise ConfigError, "Unknown #{@kind} plugin '#{type}'. Run 'gem search -rd fluentd-plugin' to find plugins"  # TODO error class
     end
 
+    def reverse_lookup(value)
+      @map.each do |k, v|
+        return k if v == value
+      end
+      nil
+    end
+
     def search(type)
       path = "#{@search_prefix}#{type}"
 
       # prefer LOAD_PATH than gems
-      files = $LOAD_PATH.map { |lp|
+      files = ($LOAD_PATH + @paths).map { |lp|
```

Comment:
Use `(@paths + $LOAD_PATH)` instead.
Fluentd's plugin pathes should override default installed plugins.